Operate import does not work for Opensearch with `UnsupportedOperationException`

### Describe the bug

<!-- A clear and concise description of what the problem/bug is about and what is the current behavior. -->
Since 67bbaf1afa679e2bf1af192780c1d1519563a62b Operate import does not work for Opensearch with `UnsupportedOperationException`

### To Reproduce

Steps to reproduce the behavior:
1. Start Operate with Opensearch
2. Deploy and start process

### Current behavior

Process is not visible in UI and following exception is logged:
```
2024-05-13T12:14:40.208+02:00 ERROR 19220 --- [       import_2] i.c.o.z.ImportJob                        : java.lang.UnsupportedOperationException: Not yet implemented

io.camunda.operate.exceptions.PersistenceException: java.lang.UnsupportedOperationException: Not yet implemented
	at io.camunda.operate.zeebeimport.AbstractImportBatchProcessor.performImport(AbstractImportBatchProcessor.java:54) ~[classes/:?]
	at io.camunda.operate.zeebeimport.ImportJob.processOneIndexBatch(ImportJob.java:161) ~[classes/:?]
	at io.camunda.operate.zeebeimport.ImportJob.call(ImportJob.java:80) ~[classes/:?]
	at io.camunda.operate.zeebeimport.opensearch.OpensearchRecordsReader.lambda$executeJob$6(OpensearchRecordsReader.java:526) ~[classes/:?]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
Caused by: java.lang.UnsupportedOperationException: Not yet implemented
	at io.camunda.operate.store.opensearch.OpensearchImportStore.getConcurrencyMode(OpensearchImportStore.java:120) ~[classes/:?]
	at io.camunda.operate.zeebeimport.processors.ListViewZeebeRecordProcessor.processProcessInstanceRecord(ListViewZeebeRecordProcessor.java:253) ~[classes/:?]
	at io.camunda.operate.zeebeimport.processors.ImportBulkProcessor.processProcessInstanceRecords(ImportBulkProcessor.java:295) ~[classes/:?]
	at io.camunda.operate.zeebeimport.processors.ImportBulkProcessor.processZeebeRecords(ImportBulkProcessor.java:132) ~[classes/:?]
	at io.camunda.operate.zeebeimport.AbstractImportBatchProcessor.lambda$performImport$0(AbstractImportBatchProcessor.java:41) ~[classes/:?]
	at io.micrometer.core.instrument.AbstractTimer.recordCallable(AbstractTimer.java:175) ~[micrometer-core-1.12.5.jar:1.12.5]
	at io.camunda.operate.zeebeimport.AbstractImportBatchProcessor.withTimer(AbstractImportBatchProcessor.java:78) ~[classes/:?]
	at io.camunda.operate.zeebeimport.AbstractImportBatchProcessor.withProcessingTimer(AbstractImportBatchProcessor.java:60) ~[classes/:?]
	at io.camunda.operate.zeebeimport.AbstractImportBatchProcessor.performImport(AbstractImportBatchProcessor.java:39) ~[classes/:?]
	... 7 more
```
<!-- Please attach screenshots, a screen recording, or a file (e.g. the BPMN/DMN/Form file) that has the problem you are describing to help us better debug the respective issue. -->

### Expected behavior

Operate shows process in UI.

<!-- A clear and concise description of what you expected to happen. -->

### Environment

<!-- Please provide details about the environment you were in when the problem occurred. -->
- OS: [e.g. MacOS]
- Browser: [e.g. chrome, safari]
- Operate Version: 8.5.1, 8.6.0-alpha1

### Additional context

<!-- Please add any other context about the problem. Here you can also provide us some data that you used while the bug happen like **json** file or specific **BPMN**. -->

---------------------------------------------------------------------------------------------

<!-- As the creator of the issue, you don't have to fill anything below this line, but the assignee will take care of this as part. -->

### Acceptance Criteria

<!-- the assignee will fill the Acceptance Criteria. -->

### Definition of Ready - Checklist

<!-- the assignee will check the DOR. -->
- [ ] The bug has been reproduced by the assignee in the environment compatible with the provided one; otherwise, the issue is closed with a comment
- [ ] The issue has a meaningful title, description, and testable acceptance criteria
- [ ] The issue has been labeled with an appropriate `Bug-area` label
- [ ] Necessary screenshots, screen recordings, or files are attached to the bug report

For UI changes required to solve the bug:

- [ ] Design input has been collected by the assignee

### Implementation

#### :mag: Root Cause Analysis

<!-- Explain the underlying cause for the issue. -->

#### :thought_balloon: Proposed Solution

<!-- Provide a high level overview of the proposed solution. (Technical details will be available in the associated PR) -->

### :point_right: Handover Dev to QA

<!--As a team, we have settled on a checklist to remind the DRI what information to provide to help the QA Engineer perform a frictionless and targeted QA test. The information requested by the checklist can be added before review/moving the ticket to the QA test column as a comment on the ticket.-->
- Changed components:
- Side effects on other components:
- Handy resources:
  BPMN/DMN models, plugins, scripts, REST API endpoints + example payload, etc :
  <!-- Add here -->
- Example projects:

<!-- Add here -->
- Commands/Steps needed to test; Versions to validate:

<!-- Add here -->
- Docker file / HELM chart : in case that it needed to be tested via docker share the version contain the fixed along with version of other services .

<!--elasticsearch: 16.2.2
identitiy:alpha3
zeebe:alpha3
Operate:alpha3
tasklist:alpha3-->
- Release version ( in which version this fixed/feature will be released):

<!-- Add here -->

### :green_book: Link to the test case

<!-- please add test case link for this bug if there is any if not after testing QA will  create a test case for it and add it here. -->

--- a/operate/importer-8_5/src/main/java/io/camunda/operate/zeebeimport/processors/DecisionEvaluationZeebeRecordProcessor.java
+++ b/operate/importer-8_5/src/main/java/io/camunda/operate/zeebeimport/processors/DecisionEvaluationZeebeRecordProcessor.java
@@ -1,9 +1,18 @@
 /*
- * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
- * one or more contributor license agreements. See the NOTICE file distributed
- * with this work for additional information regarding copyright ownership.
- * Licensed under the Zeebe Community License 1.1. You may not use this file
- * except in compliance with the Zeebe Community License 1.1.
+ * Copyright Camunda Services GmbH
+ *
+ * BY INSTALLING, DOWNLOADING, ACCESSING, USING, OR DISTRIBUTING THE SOFTWARE (“USE”), YOU INDICATE YOUR ACCEPTANCE TO AND ARE ENTERING INTO A CONTRACT WITH, THE LICENSOR ON THE TERMS SET OUT IN THIS AGREEMENT. IF YOU DO NOT AGREE TO THESE TERMS, YOU MUST NOT USE THE SOFTWARE. IF YOU ARE RECEIVING THE SOFTWARE ON BEHALF OF A LEGAL ENTITY, YOU REPRESENT AND WARRANT THAT YOU HAVE THE ACTUAL AUTHORITY TO AGREE TO THE TERMS AND CONDITIONS OF THIS AGREEMENT ON BEHALF OF SUCH ENTITY.
+ * “Licensee” means you, an individual, or the entity on whose behalf you receive the Software.
+ *
+ * Permission is hereby granted, free of charge, to the Licensee obtaining a copy of this Software and associated documentation files to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject in each case to the following conditions:
+ * Condition 1: If the Licensee distributes the Software or any derivative works of the Software, the Licensee must attach this Agreement.
+ * Condition 2: Without limiting other conditions in this Agreement, the grant of rights is solely for non-production use as defined below.
+ * "Non-production use" means any use of the Software that is not directly related to creating products, services, or systems that generate revenue or other direct or indirect economic benefits.  Examples of permitted non-production use include personal use, educational use, research, and development. Examples of prohibited production use include, without limitation, use for commercial, for-profit, or publicly accessible systems or use for commercial or revenue-generating purposes.
+ *
+ * If the Licensee is in breach of the Conditions, this Agreement, including the rights granted under it, will automatically terminate with immediate effect.
+ *
+ * SUBJECT AS SET OUT BELOW, THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * NOTHING IN THIS AGREEMENT EXCLUDES OR RESTRICTS A PARTY’S LIABILITY FOR (A) DEATH OR PERSONAL INJURY CAUSED BY THAT PARTY’S NEGLIGENCE, (B) FRAUD, OR (C) ANY OTHER LIABILITY TO THE EXTENT THAT IT CANNOT BE LAWFULLY EXCLUDED OR RESTRICTED.
  */
 package io.camunda.operate.zeebeimport.processors;
 

--- a/operate/importer-8_5/src/main/java/io/camunda/operate/zeebeimport/processors/DecisionRequirementsZeebeRecordProcessor.java
+++ b/operate/importer-8_5/src/main/java/io/camunda/operate/zeebeimport/processors/DecisionRequirementsZeebeRecordProcessor.java
@@ -1,9 +1,18 @@
 /*
- * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
- * one or more contributor license agreements. See the NOTICE file distributed
- * with this work for additional information regarding copyright ownership.
- * Licensed under the Zeebe Community License 1.1. You may not use this file
- * except in compliance with the Zeebe Community License 1.1.
+ * Copyright Camunda Services GmbH
+ *
+ * BY INSTALLING, DOWNLOADING, ACCESSING, USING, OR DISTRIBUTING THE SOFTWARE (“USE”), YOU INDICATE YOUR ACCEPTANCE TO AND ARE ENTERING INTO A CONTRACT WITH, THE LICENSOR ON THE TERMS SET OUT IN THIS AGREEMENT. IF YOU DO NOT AGREE TO THESE TERMS, YOU MUST NOT USE THE SOFTWARE. IF YOU ARE RECEIVING THE SOFTWARE ON BEHALF OF A LEGAL ENTITY, YOU REPRESENT AND WARRANT THAT YOU HAVE THE ACTUAL AUTHORITY TO AGREE TO THE TERMS AND CONDITIONS OF THIS AGREEMENT ON BEHALF OF SUCH ENTITY.
+ * “Licensee” means you, an individual, or the entity on whose behalf you receive the Software.
+ *
+ * Permission is hereby granted, free of charge, to the Licensee obtaining a copy of this Software and associated documentation files to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject in each case to the following conditions:
+ * Condition 1: If the Licensee distributes the Software or any derivative works of the Software, the Licensee must attach this Agreement.
+ * Condition 2: Without limiting other conditions in this Agreement, the grant of rights is solely for non-production use as defined below.
+ * "Non-production use" means any use of the Software that is not directly related to creating products, services, or systems that generate revenue or other direct or indirect economic benefits.  Examples of permitted non-production use include personal use, educational use, research, and development. Examples of prohibited production use include, without limitation, use for commercial, for-profit, or publicly accessible systems or use for commercial or revenue-generating purposes.
+ *
+ * If the Licensee is in breach of the Conditions, this Agreement, including the rights granted under it, will automatically terminate with immediate effect.
+ *
+ * SUBJECT AS SET OUT BELOW, THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * NOTHING IN THIS AGREEMENT EXCLUDES OR RESTRICTS A PARTY’S LIABILITY FOR (A) DEATH OR PERSONAL INJURY CAUSED BY THAT PARTY’S NEGLIGENCE, (B) FRAUD, OR (C) ANY OTHER LIABILITY TO THE EXTENT THAT IT CANNOT BE LAWFULLY EXCLUDED OR RESTRICTED.
  */
 package io.camunda.operate.zeebeimport.processors;
 

--- a/operate/importer-8_5/src/main/java/io/camunda/operate/zeebeimport/processors/DecisionZeebeRecordProcessor.java
+++ b/operate/importer-8_5/src/main/java/io/camunda/operate/zeebeimport/processors/DecisionZeebeRecordProcessor.java
@@ -1,9 +1,18 @@
 /*
- * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
- * one or more contributor license agreements. See the NOTICE file distributed
- * with this work for additional information regarding copyright ownership.
- * Licensed under the Zeebe Community License 1.1. You may not use this file
- * except in compliance with the Zeebe Community License 1.1.
+ * Copyright Camunda Services GmbH
+ *
+ * BY INSTALLING, DOWNLOADING, ACCESSING, USING, OR DISTRIBUTING THE SOFTWARE (“USE”), YOU INDICATE YOUR ACCEPTANCE TO AND ARE ENTERING INTO A CONTRACT WITH, THE LICENSOR ON THE TERMS SET OUT IN THIS AGREEMENT. IF YOU DO NOT AGREE TO THESE TERMS, YOU MUST NOT USE THE SOFTWARE. IF YOU ARE RECEIVING THE SOFTWARE ON BEHALF OF A LEGAL ENTITY, YOU REPRESENT AND WARRANT THAT YOU HAVE THE ACTUAL AUTHORITY TO AGREE TO THE TERMS AND CONDITIONS OF THIS AGREEMENT ON BEHALF OF SUCH ENTITY.
+ * “Licensee” means you, an individual, or the entity on whose behalf you receive the Software.
+ *
+ * Permission is hereby granted, free of charge, to the Licensee obtaining a copy of this Software and associated documentation files to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject in each case to the following conditions:
+ * Condition 1: If the Licensee distributes the Software or any derivative works of the Software, the Licensee must attach this Agreement.
+ * Condition 2: Without limiting other conditions in this Agreement, the grant of rights is solely for non-production use as defined below.
+ * "Non-production use" means any use of the Software that is not directly related to creating products, services, or systems that generate revenue or other direct or indirect economic benefits.  Examples of permitted non-production use include personal use, educational use, research, and development. Examples of prohibited production use include, without limitation, use for commercial, for-profit, or publicly accessible systems or use for commercial or revenue-generating purposes.
+ *
+ * If the Licensee is in breach of the Conditions, this Agreement, including the rights granted under it, will automatically terminate with immediate effect.
+ *
+ * SUBJECT AS SET OUT BELOW, THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * NOTHING IN THIS AGREEMENT EXCLUDES OR RESTRICTS A PARTY’S LIABILITY FOR (A) DEATH OR PERSONAL INJURY CAUSED BY THAT PARTY’S NEGLIGENCE, (B) FRAUD, OR (C) ANY OTHER LIABILITY TO THE EXTENT THAT IT CANNOT BE LAWFULLY EXCLUDED OR RESTRICTED.
  */
 package io.camunda.operate.zeebeimport.processors;
 

--- a/operate/importer-8_5/src/main/java/io/camunda/operate/zeebeimport/processors/SequenceFlowZeebeRecordProcessor.java
+++ b/operate/importer-8_5/src/main/java/io/camunda/operate/zeebeimport/processors/SequenceFlowZeebeRecordProcessor.java
@@ -1,9 +1,18 @@
 /*
- * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
- * one or more contributor license agreements. See the NOTICE file distributed
- * with this work for additional information regarding copyright ownership.
- * Licensed under the Zeebe Community License 1.1. You may not use this file
- * except in compliance with the Zeebe Community License 1.1.
+ * Copyright Camunda Services GmbH
+ *
+ * BY INSTALLING, DOWNLOADING, ACCESSING, USING, OR DISTRIBUTING THE SOFTWARE (“USE”), YOU INDICATE YOUR ACCEPTANCE TO AND ARE ENTERING INTO A CONTRACT WITH, THE LICENSOR ON THE TERMS SET OUT IN THIS AGREEMENT. IF YOU DO NOT AGREE TO THESE TERMS, YOU MUST NOT USE THE SOFTWARE. IF YOU ARE RECEIVING THE SOFTWARE ON BEHALF OF A LEGAL ENTITY, YOU REPRESENT AND WARRANT THAT YOU HAVE THE ACTUAL AUTHORITY TO AGREE TO THE TERMS AND CONDITIONS OF THIS AGREEMENT ON BEHALF OF SUCH ENTITY.
+ * “Licensee” means you, an individual, or the entity on whose behalf you receive the Software.
+ *
+ * Permission is hereby granted, free of charge, to the Licensee obtaining a copy of this Software and associated documentation files to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject in each case to the following conditions:
+ * Condition 1: If the Licensee distributes the Software or any derivative works of the Software, the Licensee must attach this Agreement.
+ * Condition 2: Without limiting other conditions in this Agreement, the grant of rights is solely for non-production use as defined below.
+ * "Non-production use" means any use of the Software that is not directly related to creating products, services, or systems that generate revenue or other direct or indirect economic benefits.  Examples of permitted non-production use include personal use, educational use, research, and development. Examples of prohibited production use include, without limitation, use for commercial, for-profit, or publicly accessible systems or use for commercial or revenue-generating purposes.
+ *
+ * If the Licensee is in breach of the Conditions, this Agreement, including the rights granted under it, will automatically terminate with immediate effect.
+ *
+ * SUBJECT AS SET OUT BELOW, THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * NOTHING IN THIS AGREEMENT EXCLUDES OR RESTRICTS A PARTY’S LIABILITY FOR (A) DEATH OR PERSONAL INJURY CAUSED BY THAT PARTY’S NEGLIGENCE, (B) FRAUD, OR (C) ANY OTHER LIABILITY TO THE EXTENT THAT IT CANNOT BE LAWFULLY EXCLUDED OR RESTRICTED.
  */
 package io.camunda.operate.zeebeimport.processors;
 

--- a/operate/importer-8_5/src/main/java/io/camunda/operate/zeebeimport/processors/UserTaskZeebeRecordProcessor.java
+++ b/operate/importer-8_5/src/main/java/io/camunda/operate/zeebeimport/processors/UserTaskZeebeRecordProcessor.java
@@ -1,9 +1,18 @@
 /*
- * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
- * one or more contributor license agreements. See the NOTICE file distributed
- * with this work for additional information regarding copyright ownership.
- * Licensed under the Zeebe Community License 1.1. You may not use this file
- * except in compliance with the Zeebe Community License 1.1.
+ * Copyright Camunda Services GmbH
+ *
+ * BY INSTALLING, DOWNLOADING, ACCESSING, USING, OR DISTRIBUTING THE SOFTWARE (“USE”), YOU INDICATE YOUR ACCEPTANCE TO AND ARE ENTERING INTO A CONTRACT WITH, THE LICENSOR ON THE TERMS SET OUT IN THIS AGREEMENT. IF YOU DO NOT AGREE TO THESE TERMS, YOU MUST NOT USE THE SOFTWARE. IF YOU ARE RECEIVING THE SOFTWARE ON BEHALF OF A LEGAL ENTITY, YOU REPRESENT AND WARRANT THAT YOU HAVE THE ACTUAL AUTHORITY TO AGREE TO THE TERMS AND CONDITIONS OF THIS AGREEMENT ON BEHALF OF SUCH ENTITY.
+ * “Licensee” means you, an individual, or the entity on whose behalf you receive the Software.
+ *
+ * Permission is hereby granted, free of charge, to the Licensee obtaining a copy of this Software and associated documentation files to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject in each case to the following conditions:
+ * Condition 1: If the Licensee distributes the Software or any derivative works of the Software, the Licensee must attach this Agreement.
+ * Condition 2: Without limiting other conditions in this Agreement, the grant of rights is solely for non-production use as defined below.
+ * "Non-production use" means any use of the Software that is not directly related to creating products, services, or systems that generate revenue or other direct or indirect economic benefits.  Examples of permitted non-production use include personal use, educational use, research, and development. Examples of prohibited production use include, without limitation, use for commercial, for-profit, or publicly accessible systems or use for commercial or revenue-generating purposes.
+ *
+ * If the Licensee is in breach of the Conditions, this Agreement, including the rights granted under it, will automatically terminate with immediate effect.
+ *
+ * SUBJECT AS SET OUT BELOW, THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * NOTHING IN THIS AGREEMENT EXCLUDES OR RESTRICTS A PARTY’S LIABILITY FOR (A) DEATH OR PERSONAL INJURY CAUSED BY THAT PARTY’S NEGLIGENCE, (B) FRAUD, OR (C) ANY OTHER LIABILITY TO THE EXTENT THAT IT CANNOT BE LAWFULLY EXCLUDED OR RESTRICTED.
  */
 package io.camunda.operate.zeebeimport.processors;
 

--- a/operate/qa/integration-tests/src/test/java/io/camunda/operate/util/ZeebeRecordTestUtil.java
+++ b/operate/qa/integration-tests/src/test/java/io/camunda/operate/util/ZeebeRecordTestUtil.java
@@ -205,6 +205,7 @@ public abstract class ZeebeRecordTestUtil {
         .withName(var.getName())
         .withValue(var.getFullValue())
         .withProcessInstanceKey(var.getProcessInstanceKey())
+        .withBpmnProcessId(var.getBpmnProcessId())
         .withScopeKey(var.getScopeKey())
         .withTenantId(var.getTenantId());
     if (recordValueBuilderFunction != null) {

--- a/operate/schema/src/main/java/io/camunda/operate/store/opensearch/OpensearchImportStore.java
+++ b/operate/schema/src/main/java/io/camunda/operate/store/opensearch/OpensearchImportStore.java
@@ -16,6 +16,7 @@
  */
 package io.camunda.operate.store.opensearch;
 
+import static io.camunda.operate.schema.indices.ImportPositionIndex.META_CONCURRENCY_MODE;
 import static io.camunda.operate.store.opensearch.dsl.QueryDSL.and;
 import static io.camunda.operate.store.opensearch.dsl.QueryDSL.term;
 import static io.camunda.operate.store.opensearch.dsl.RequestDSL.searchRequestBuilder;
@@ -25,15 +26,19 @@ import io.camunda.operate.Metrics;
 import io.camunda.operate.conditions.OpensearchCondition;
 import io.camunda.operate.entities.meta.ImportPositionEntity;
 import io.camunda.operate.property.OperateProperties;
+import io.camunda.operate.schema.IndexMapping;
 import io.camunda.operate.schema.indices.ImportPositionIndex;
 import io.camunda.operate.store.ImportStore;
 import io.camunda.operate.store.opensearch.client.sync.RichOpenSearchClient;
 import io.camunda.operate.util.Either;
 import java.io.IOException;
 import java.util.List;
+import java.util.Map;
 import java.util.concurrent.Callable;
 import java.util.function.Function;
+import org.opensearch.client.json.JsonData;
 import org.opensearch.client.opensearch.core.BulkRequest;
+import org.opensearch.client.opensearch.indices.PutMappingRequest;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
@@ -55,7 +60,7 @@ public class OpensearchImportStore implements ImportStore {
   @Autowired private OperateProperties operateProperties;
 
   @Override
-  public ImportPositionEntity getImportPositionByAliasAndPartitionId(String alias, int partitionId)
+  public ImportPositionEntity getImportPositionByAliasAndPartitionId(final String alias, final int partitionId)
       throws IOException {
     final var searchRequestBuilder =
         searchRequestBuilder(importPositionType.getAlias())
@@ -84,7 +89,7 @@ public class OpensearchImportStore implements ImportStore {
 
   @Override
   public Either<Throwable, Boolean> updateImportPositions(
-      List<ImportPositionEntity> positions, List<ImportPositionEntity> postImportPositions) {
+      final List<ImportPositionEntity> positions, final List<ImportPositionEntity> postImportPositions) {
     if (positions.isEmpty() && postImportPositions.isEmpty()) {
       return Either.right(true);
     } else {
@@ -110,25 +115,15 @@ public class OpensearchImportStore implements ImportStore {
     }
   }
 
-  @Override
-  public void setConcurrencyMode(final boolean concurrencyMode) {
-    throw new UnsupportedOperationException("Not yet implemented");
-  }
-
-  @Override
-  public boolean getConcurrencyMode() {
-    throw new UnsupportedOperationException("Not yet implemented");
-  }
-
   private void withImportPositionTimer(final Callable<Void> action) throws Exception {
     metrics.getTimer(Metrics.TIMER_NAME_IMPORT_POSITION_UPDATE).recordCallable(action);
   }
 
-  private <R> void addPositions(
+private <R> void addPositions(
       final BulkRequest.Builder bulkRequestBuilder,
       final List<ImportPositionEntity> positions,
-      Function<ImportPositionEntity, R> entityProducer) {
-    for (ImportPositionEntity position : positions) {
+      final Function<ImportPositionEntity, R> entityProducer) {
+    for (final ImportPositionEntity position : positions) {
       bulkRequestBuilder.operations(
           op ->
               op.update(
@@ -139,8 +134,7 @@ public class OpensearchImportStore implements ImportStore {
                           .document(entityProducer.apply(position))));
     }
   }
-
-  record ImportPositionUpdate(
+    record ImportPositionUpdate(
       String id,
       String aliasName,
       String indexName,
@@ -159,12 +153,37 @@ public class OpensearchImportStore implements ImportStore {
           position.getPostImporterPosition(),
           position.getSequence());
     }
+  }@Override
+  public void setConcurrencyMode(final boolean concurrencyMode) {
+    final String indexName = importPositionType.getFullQualifiedName();
+    LOGGER.debug("Meta field will be updated. Index name: {}. ");
+    final PutMappingRequest request =
+        new PutMappingRequest.Builder()
+            .index(indexName)
+            .meta(META_CONCURRENCY_MODE, JsonData.of(concurrencyMode))
+            .build();
+    richOpenSearchClient.index().putMapping(request);
   }
 
   record PostImportPositionUpdate(Long postImporterPosition) {
     public static PostImportPositionUpdate fromImportPositionEntity(
         final ImportPositionEntity position) {
       return new PostImportPositionUpdate(position.getPostImporterPosition());
+    }
+  }
+
+
+  @Override
+  public boolean getConcurrencyMode() {
+    final String indexName = importPositionType.getFullQualifiedName();
+    final Map<String, IndexMapping> indexMappings = richOpenSearchClient.index()
+        .getIndexMappings(indexName);
+    if (indexMappings.get(indexName).getMetaProperties() == null) {
+      return false;
+    } else {
+      final Object concurrencyMode =
+          indexMappings.get(indexName).getMetaProperties().get(META_CONCURRENCY_MODE);
+      return concurrencyMode == null ? false : (boolean) concurrencyMode;
     }
   }
 }

--- a/operate/schema/src/main/java/io/camunda/operate/store/opensearch/OpensearchImportStore.java
+++ b/operate/schema/src/main/java/io/camunda/operate/store/opensearch/OpensearchImportStore.java
@@ -60,8 +60,8 @@ public class OpensearchImportStore implements ImportStore {
   @Autowired private OperateProperties operateProperties;
 
   @Override
-  public ImportPositionEntity getImportPositionByAliasAndPartitionId(final String alias, final int partitionId)
-      throws IOException {
+  public ImportPositionEntity getImportPositionByAliasAndPartitionId(
+      final String alias, final int partitionId) throws IOException {
     final var searchRequestBuilder =
         searchRequestBuilder(importPositionType.getAlias())
             .size(10)
@@ -89,7 +89,8 @@ public class OpensearchImportStore implements ImportStore {
 
   @Override
   public Either<Throwable, Boolean> updateImportPositions(
-      final List<ImportPositionEntity> positions, final List<ImportPositionEntity> postImportPositions) {
+      final List<ImportPositionEntity> positions,
+      final List<ImportPositionEntity> postImportPositions) {
     if (positions.isEmpty() && postImportPositions.isEmpty()) {
       return Either.right(true);
     } else {
@@ -119,7 +120,7 @@ public class OpensearchImportStore implements ImportStore {
     metrics.getTimer(Metrics.TIMER_NAME_IMPORT_POSITION_UPDATE).recordCallable(action);
   }
 
-private <R> void addPositions(
+  private <R> void addPositions(
       final BulkRequest.Builder bulkRequestBuilder,
       final List<ImportPositionEntity> positions,
       final Function<ImportPositionEntity, R> entityProducer) {
@@ -134,7 +135,34 @@ private <R> void addPositions(
                           .document(entityProducer.apply(position))));
     }
   }
-    record ImportPositionUpdate(
+
+  @Override
+  public void setConcurrencyMode(final boolean concurrencyMode) {
+    final String indexName = importPositionType.getFullQualifiedName();
+    LOGGER.debug("Meta field will be updated. Index name: {}. ");
+    final PutMappingRequest request =
+        new PutMappingRequest.Builder()
+            .index(indexName)
+            .meta(META_CONCURRENCY_MODE, JsonData.of(concurrencyMode))
+            .build();
+    richOpenSearchClient.index().putMapping(request);
+  }
+
+  @Override
+  public boolean getConcurrencyMode() {
+    final String indexName = importPositionType.getFullQualifiedName();
+    final Map<String, IndexMapping> indexMappings =
+        richOpenSearchClient.index().getIndexMappings(indexName);
+    if (indexMappings.get(indexName).getMetaProperties() == null) {
+      return false;
+    } else {
+      final Object concurrencyMode =
+          indexMappings.get(indexName).getMetaProperties().get(META_CONCURRENCY_MODE);
+      return concurrencyMode != null && (boolean) concurrencyMode;
+    }
+  }
+
+  record ImportPositionUpdate(
       String id,
       String aliasName,
       String indexName,
@@ -153,37 +181,12 @@ private <R> void addPositions(
           position.getPostImporterPosition(),
           position.getSequence());
     }
-  }@Override
-  public void setConcurrencyMode(final boolean concurrencyMode) {
-    final String indexName = importPositionType.getFullQualifiedName();
-    LOGGER.debug("Meta field will be updated. Index name: {}. ");
-    final PutMappingRequest request =
-        new PutMappingRequest.Builder()
-            .index(indexName)
-            .meta(META_CONCURRENCY_MODE, JsonData.of(concurrencyMode))
-            .build();
-    richOpenSearchClient.index().putMapping(request);
   }
 
   record PostImportPositionUpdate(Long postImporterPosition) {
     public static PostImportPositionUpdate fromImportPositionEntity(
         final ImportPositionEntity position) {
       return new PostImportPositionUpdate(position.getPostImporterPosition());
-    }
-  }
-
-
-  @Override
-  public boolean getConcurrencyMode() {
-    final String indexName = importPositionType.getFullQualifiedName();
-    final Map<String, IndexMapping> indexMappings = richOpenSearchClient.index()
-        .getIndexMappings(indexName);
-    if (indexMappings.get(indexName).getMetaProperties() == null) {
-      return false;
-    } else {
-      final Object concurrencyMode =
-          indexMappings.get(indexName).getMetaProperties().get(META_CONCURRENCY_MODE);
-      return concurrencyMode == null ? false : (boolean) concurrencyMode;
     }
   }
 }

--- a/operate/schema/src/main/java/io/camunda/operate/store/opensearch/client/sync/OpenSearchIndexOperations.java
+++ b/operate/schema/src/main/java/io/camunda/operate/store/opensearch/client/sync/OpenSearchIndexOperations.java
@@ -28,6 +28,8 @@ import java.util.*;
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
 import org.elasticsearch.rest.RestStatus;
+import org.opensearch.client.json.JsonData;
+import org.opensearch.client.json.JsonpDeserializer;
 import org.opensearch.client.opensearch.OpenSearchClient;
 import org.opensearch.client.opensearch._types.OpenSearchException;
 import org.opensearch.client.opensearch._types.mapping.Property;
@@ -247,11 +249,21 @@ public class OpenSearchIndexOperations extends OpenSearchRetryOperation {
                 indexMapping.getValue().mappings().dynamic() == null
                     ? null
                     : indexMapping.getValue().mappings().dynamic().name();
+            final Map<String, Object> metaFields = new HashMap<>();
+            if (indexMapping.getValue().mappings().meta() != null) {
+              for (final Map.Entry<String, JsonData> entry :
+                  indexMapping.getValue().mappings().meta().entrySet()) {
+                metaFields.put(
+                    entry.getKey(),
+                    entry.getValue().deserialize(JsonpDeserializer.booleanDeserializer()));
+              }
+            }
             final IndexMapping mapping =
                 new IndexMapping()
                     .setIndexName(indexMapping.getKey())
                     .setDynamic(dynamic)
-                    .setProperties(properties);
+                    .setProperties(properties)
+                    .setMetaProperties(metaFields);
             mappings.put(indexMapping.getKey(), mapping);
           }
           return mappings;

--- a/operate/schema/src/test/java/io/camunda/operate/schema/ImportStoreIT.java
+++ b/operate/schema/src/test/java/io/camunda/operate/schema/ImportStoreIT.java
@@ -22,10 +22,12 @@ import io.camunda.operate.JacksonConfig;
 import io.camunda.operate.Metrics;
 import io.camunda.operate.conditions.DatabaseInfo;
 import io.camunda.operate.connect.ElasticsearchConnector;
+import io.camunda.operate.connect.OpensearchConnector;
 import io.camunda.operate.connect.OperateDateTimeFormatter;
 import io.camunda.operate.property.OperateProperties;
 import io.camunda.operate.schema.elasticsearch.ElasticsearchSchemaManager;
 import io.camunda.operate.schema.indices.ImportPositionIndex;
+import io.camunda.operate.schema.opensearch.OpensearchSchemaManager;
 import io.camunda.operate.schema.util.SchemaTestHelper;
 import io.camunda.operate.schema.util.TestTemplate;
 import io.camunda.operate.schema.util.elasticsearch.ElasticsearchSchemaTestHelper;
@@ -34,6 +36,7 @@ import io.camunda.operate.store.ImportStore;
 import io.camunda.operate.store.elasticsearch.ElasticsearchImportStore;
 import io.camunda.operate.store.elasticsearch.ElasticsearchTaskStore;
 import io.camunda.operate.store.elasticsearch.RetryElasticsearchClient;
+import io.camunda.operate.store.opensearch.OpensearchImportStore;
 import io.micrometer.core.instrument.MeterRegistry;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
@@ -49,12 +52,15 @@ import org.springframework.test.context.ContextConfiguration;
       TestTemplate.class,
       ElasticsearchSchemaManager.class,
       ElasticsearchImportStore.class,
+      OpensearchSchemaManager.class,
+      OpensearchImportStore.class,
       OpenSearchSchemaTestHelper.class,
       ElasticsearchSchemaTestHelper.class,
       DatabaseInfo.class,
       OperateProperties.class,
       RetryElasticsearchClient.class,
       ElasticsearchConnector.class,
+      OpensearchConnector.class,
       ElasticsearchTaskStore.class,
       JacksonConfig.class,
       OperateDateTimeFormatter.class,

